### PR TITLE
Delete module role from existing person

### PR DESCRIPTION
**Pull request by lithops-zty**
A follow-up of PR #129.

syntax:
`edit [index] /r-(MODULECODE[-ROLETYPE])+`
e.g. edit 1 /r-cs1101s ma1522-ta

Follow a sequential check, invalid module-role-to-delete will be ignored instead of raising exception. This behavior is subject to further discussion.

Moreover, the edit result message now shows what fields are edited.
image

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/edit/EditCommand.java [hunk 1]
```diff
@@ -112,6 +115,47 @@ private static Person createEditedPerson(Person personToEdit, EditPersonDescript
                 updatedTags, updatedModuleRoleMap);
     }
 
+    /**
+     * Returns a description of the changes made to the person.
+     */
+    public static String getChangeDescription(Person personBefore, Person personAfter) {
+        StringBuilder changeDescription = new StringBuilder("Changes made: \n");
```
Comment (MAOXIONGKAI) [inline src/main/java/seedu/address/logic/commands/edit/EditCommand.java:122]: Would it be better to change `Changes made` to `Change(s) made`?

File: src/main/java/seedu/address/logic/commands/edit/EditModuleRoleOperation.java [hunk 2]
```diff
@@ -33,4 +37,41 @@ public abstract class EditModuleRoleOperation {
     public static boolean isValidModuleRoleOperation(String operation) {
         return operation.toLowerCase().matches(VALIDATION_REGEX);
     }
+
+    /**
+     * Returns a description of the change in module roles.
+     * @param moduleRoleMapBefore The module role map before the change.
+     * @param moduleRoleMapAfter The module role map after the change.
+     * @return A description of the change in module roles.
```
Comment (MAOXIONGKAI) [inline src/main/java/seedu/address/logic/commands/edit/EditModuleRoleOperation.java:45]: I think can include an example over here for easier visualization, you can refer to `getModuleCodeString` method Javadoc to see how it explains what it does

File: src/main/java/seedu/address/logic/commands/edit/EditModuleRoleOperation.java [hunk 3]
```diff
@@ -33,4 +37,41 @@ public abstract class EditModuleRoleOperation {
...
+                hasDeleted = true;
+                stringBuilderDeleted.append(moduleRolePair).append(" ");
+            }
+        }
+        List<String> finalDescription = new ArrayList<>();
+        if (hasAdded) {
+            finalDescription.add("Module role added: " + stringBuilderAdded.toString().strip());
+        }
+        if (hasDeleted) {
+            finalDescription.add("Module role deleted: " + stringBuilderDeleted.toString().strip());
```
Comment (MAOXIONGKAI) [inline src/main/java/seedu/address/logic/commands/edit/EditModuleRoleOperation.java:72]: Would it better to changes the messages to `Module role(s) added` and `Module role(s) deleted` instead?

File: src/test/java/seedu/address/logic/commands/edit/EditCommandTest.java [hunk 4]
```diff
@@ -206,4 +224,35 @@ public void toStringMethod() {
...
+        // changed all fields
+        Person person = new PersonBuilder().withName(VALID_NAME_BOB).withPhone(VALID_PHONE_BOB)
+                .withEmail(VALID_EMAIL_AMY).withEmptyAddress()
+                .withTags(VALID_TAG_HUSBAND).withModuleRoleMap(new ModuleCode("MA1522"), RoleType.TUTOR).build();
+
+        Person editedPerson = new PersonBuilder().withName(VALID_NAME_BETTY).withPhone(VALID_PHONE_BETTY)
+                .withEmail(VALID_EMAIL_BETTY).withAddress(VALID_ADDRESS_AMY)
+                .withTags(VALID_TAG_FRIEND).withModuleRoleMap(new ModuleCode("MA2001"), RoleType.PROFESSOR).build();
+
+        String expected = "Changes made: "
```
Comment (MAOXIONGKAI) [inline src/test/java/seedu/address/logic/commands/edit/EditCommandTest.java:238]: If you decide to change this to `Change(s)` made, remember to update such test cases

**Review by MAOXIONGKAI — COMMENTED** [review]
Overall looks good to me! 👍 I think only need to include some examples in some of the Javadoc for better clarity, and improve on some existing messages to the user, then can just merge already

File: src/main/java/seedu/address/logic/commands/edit/EditModuleRoleOperation.java (hunk 2 above)
Comment (lithops-zty) [reply src/main/java/seedu/address/logic/commands/edit/EditModuleRoleOperation.java:45]: good suggestion, added!

**Comment by lithops-zty** [comment]
@MAOXIONGKAI thanks for the review! I've updated code based on your suggestions

**Review by MAOXIONGKAI — APPROVED** [review]
LGTM! However, since the current implementation does not support more flexible usage(Let CS1101S denotes all roles instead of just student) and currently there's no quick fix to this, I will create an issue on this for v1.5 and think about what to do to fix this. For now, can just merge this

**Pull request merged**